syft/format/internal/testutil/snapshot.go

```diff
@@ -62,6 +58,13 @@ func AssertEncoderAgainstGoldenSnapshot(t *testing.T, cfg EncoderSnapshotTestCon
 		return
 	}
 
+	var expected []byte
```

Comment:
I don't understand why this code moved down. It seems like this change has no effect, which makes me think I'm missing something.

Reply:
I should have explicitly labeled this change (I can split this into another PR if needed). The first iteration of this PR was heavy on snapshot testing, which after adding I backed out the change and added more explicit tests. In adding these snapshots I was surprised to see that I couldn't add any snapshots! This fixes that bug, where the snapshot needed to exist in order to update it, which is not correct.